extensions/systemd/ConsumeJournald.cpp

```diff
@@ -0,0 +1,262 @@
+    ->withAllowableValues<std::string>({PAYLOAD_FORMAT_RAW, PAYLOAD_FORMAT_SYSLOG})
+    ->isRequired(true)
+    ->build();
+
+const core::Property ConsumeJournald::IncludeTimestamp = core::PropertyBuilder::createProperty("Include Timestamp")
+    ->withDescription("Include message timestamp in the 'timestamp' attribute.")
+    ->withDefaultValue<bool>(true)
+    ->isRequired(true)
+    ->build();
+
+const core::Property ConsumeJournald::JournalType = core::PropertyBuilder::createProperty("Journal Type")
+    ->withDescription("Type of journal to consume.")
+    ->withDefaultValue<std::string>(JOURNAL_TYPE_SYSTEM)
+    ->withAllowableValues<std::string>({JOURNAL_TYPE_USER, JOURNAL_TYPE_SYSTEM, JOURNAL_TYPE_BOTH})
+    ->isRequired(true)
+    ->build();
+
+const core::Property ConsumeJournald::ProcessOldMessages = core::PropertyBuilder::createProperty("Process Old Messages")
+    ->withDescription("Process events created before the first usage (schedule) of the processor instance.")
+    ->withDefaultValue<bool>(false)
+    ->isRequired(true)
+    ->build();
+
+const core::Property ConsumeJournald::TimestampFormat = core::PropertyBuilder::createProperty("Timestamp Format")
+    ->withDescription("Format string to use when creating the timestamp attribute or writing messages in the syslog format.")
+    ->withDefaultValue("%x %X %Z")
+    ->isRequired(true)
+    ->build();
+
+ConsumeJournald::ConsumeJournald(const std::string &name, const utils::Identifier &id, std::unique_ptr<libwrapper::LibWrapper>&& libwrapper)
+    :core::Processor{name, id}, libwrapper_{std::move(libwrapper)}
+{}
+
+void ConsumeJournald::initialize() {
+  setSupportedProperties({BatchSize, PayloadFormat, IncludeTimestamp, JournalType, ProcessOldMessages, TimestampFormat});
+  setSupportedRelationships({Success});
+
+  worker_ = utils::make_unique<Worker>();
+}
+
+void ConsumeJournald::notifyStop() {
+  bool running = true;
+  if (!running_.compare_exchange_strong(running, false, std::memory_order_acq_rel) || !journal_) return;
+  worker_->enqueue([this] {
+    journal_ = nullptr;
+  }).get();
+  worker_ = nullptr;
+}
+
+void ConsumeJournald::onSchedule(core::ProcessContext* const context, core::ProcessSessionFactory* const sessionFactory) {
+  gsl_Expects(context && sessionFactory && !running_ && worker_);
+  using JournalTypeEnum = systemd::JournalType;
+
+  const auto parse_payload_format = [](const std::string& property_value) -> utils::optional<systemd::PayloadFormat> {
+    if (utils::StringUtils::equalsIgnoreCase(property_value, PAYLOAD_FORMAT_RAW)) return systemd::PayloadFormat::Raw;
+    if (utils::StringUtils::equalsIgnoreCase(property_value, PAYLOAD_FORMAT_SYSLOG)) return systemd::PayloadFormat::Syslog;
+    return utils::nullopt;
+  };
+  const auto parse_journal_type = [](const std::string& property_value) -> utils::optional<JournalTypeEnum> {
+    if (utils::StringUtils::equalsIgnoreCase(property_value, JOURNAL_TYPE_USER)) return JournalTypeEnum::User;
```

Comment:
it seems that `ConsumeKafka` is the only other processor supporting case-insensitive property values, we should keep property values case sensitive until there is apt reason to decide otherwise

Reply:
changed to case-sensitive in f936b60